mne_bids/read.py

```diff
@@ -268,6 +268,29 @@ def _handle_info_reading(sidecar_fname, raw, verbose=None):
                              "Sidecar JSON is -> {} ".format(line_freq))
 
     raw.info["line_freq"] = line_freq
+
+    # get cHPI info
+    chpi = sidecar_json.get('ContinuousHeadLocalization')
+    if chpi is None:
+        # no cHPI info in the sidecar – leave raw.info unchanged
+        pass
+    elif chpi is True:
+        hpi_freqs = sidecar_json['HeadCoilFrequency']
+        hpi_freqs_data, _, _ = mne.chpi.get_chpi_info(raw.info)
+        if not np.allclose(hpi_freqs, hpi_freqs_data):
+            raise ValueError(
+                f'The cHPI coil frequencies in the sidecar file '
+                f'{sidecar_fname}:\n    {hpi_freqs}\ndiffer from what is '
+                f'stored in the raw data:\n    {hpi_freqs_data}\n'
+                f'Cannot proceed.'
+            )
+    else:
+        if raw.info['hpi_subsystem']:
+            logger.info('Dropping cHPI information stored in raw data, '
+                        'following specification in sidecar file')
+        raw.info['hpi_subsystem'] = None
+        raw.info['hpi_meas'] = []
+
```

Comment:
I don't get this part:

- when chpi info is present in BIDS but conflicts with raw (either absent in raw, or different in raw), you raise
- when chpi info is absent in BIDS, but present in raw, you remove the info from raw

This would make more sense to me:
- when chpi info is present in BIDS: use it ... if it conflicts with raw, send a warning, but overwrite raw with BIDS info
- when chpi info is absent in BIDS, but present in raw: send warning that BIDS has no info ... but we'll be using what's in raw

what was your reasoning for these cases?

Reply:
> I don't get this part:
> 
> * when chpi info is present in BIDS but conflicts with raw (either absent in raw, or different in raw), you raise

This is correct.

> * when chpi info is absent in BIDS, but present in raw, you remove the info from raw

This is not exactly correct: only if there **is** a `ContinuousHeadLocalization` setting in the sidecar **and** it is set to `false`, we drop the info from the raw.

If we dropped the info whenever `ContinuousHeadLocalization` was missing, we'd just throw away some data that's actually there … Does that make sense?

> 
> This would make more sense to me:
> 
> * when chpi info is present in BIDS: use it ... if it conflicts with raw, send a warning, but overwrite raw with BIDS info

The issue here is that the cHPI info stored in BIDS is not sufficient to restore much of anything that would actually be needed to to any kind of cHPI-related processing. The sidecar really stores "there was an cHPI in use" and "those were the frequency bands that the cHPI coils occupied". Really just very "meta" :)

Reply:
I see, thanks for the explanation :thinking: so BIDS would need additional metadata fields for cHPI-related data if one wants to **really** make use of it?

In any case, this makes sense to me now, +1 for merge